clients/cli/cmd/internal/push.go

```diff
@@ -209,13 +212,29 @@ func (source *Source) Push(client *phrase.APIClient, waitForResults bool, branch
 			fmt.Fprintln(os.Stderr, strings.Repeat("-", 10))
 		}
 	}
-	if !noErrors {
+	if noErrors {
+		if waitForResults && cleanup {
```

Comment:
Maybe you can show an error when the `cleanup` is passed but `wait` is missing? Otherwise it might cause some confusion why the cleanup didn't happen